sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -609,7 +611,17 @@ struct pi_command_list_info_t {
   // TODO: use this for optimizing events in the same command-list, e.g.
   // only have last one visible to the host.
   std::vector<pi_event> EventList{};
-  size_t size() const { return EventList.size(); }
+
+  // List of kernels in this command list associated with discarded events.
+  std::list<pi_kernel> Kernels;
+
+  // List of dependent events associated with discarded events in this command
+  // list.
+  std::list<_pi_ze_event_list_t> WaitLists;
+
+  size_t NumDiscardedEvents = 0;
+
+  size_t size() const { return EventList.size() + NumDiscardedEvents; }
```

Comment:
why do we need to handle-discarded events with completely separate data-structures? Can you extend the 

> std::vector<pi_event> EventList{};

for discarded events be just a special case of that existing list?

Reply:
Whole idea of the patch was avoiding creation/deletion of pi_event objects for discarded events.  That's why it is implemented this way.

Reply:
Really? I thought the  main idea is to not create excessive L0 events, which you correctly tell in the title:

> Reuse discarded L0 events in scope of command list

I don't think it is important (at least now) to not create PI events, which are just a "thin" wrapper that was never blamed to be a problem.